File: Practice_team06/Models/Seat.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Practice_team06.Models;

public partial class Seat
{
    public int Id { get; set; }
    [Required]
    public int HallId { get; set; }
    [Required]
    [Range(1, 200)]
    public short RowNumber { get; set; }
    [Required]
    [Range(1, 200)]
    public short SeatNumber { get; set; }
    [Range(0.1, 5.0)]
    public decimal? PriceModifier { get; set; }
    
    public SeatType SeatType { get; set; }
    
    public SeatStatus SeatStatus { get; set; }

    public virtual Hall Hall { get; set; } = null!;

    public virtual ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
}
public enum SeatType { Standard, VIP }

public enum SeatStatus { Free, Reserved, Sold }

File: cinema-frontend/src/types/seat.ts
export const SeatType = {
    Standard: 0,
    VIP: 1,
} as const;

export type SeatType = typeof SeatType[keyof typeof SeatType];

export interface SeatDto {
    id: number;
    hallId: number;
    rowNumber: number;
    seatNumber: number;
    priceModifier: number;
    seatType: SeatType;
}

export interface SessionSeatDto {
    seatId: number;
    rowNumber: number;
    seatNumber: number;
    type: SeatType;
    isAvailable: boolean;
    price: number; // decimal -> number
}

export interface UpdateSeatDto {
    seatType: SeatType;
    priceModifier: number;
}